parsl/tests/test_python_apps/test_pluggable_future_resolution.py

```diff
@@ -29,14 +30,41 @@ def test_simple_pos_arg():
 
 
 @parsl.python_app
-def b_first(x: Sequence[int]):
+def b_first(x: Iterable[int]):
     return x[0] + 1
 
 
 @pytest.mark.local
 def test_tuple_pos_arg():
     e = Event()
-    s = (a(e), )
+    s = (a(e),)
     f_b = b_first(s)
     e.set()
     assert f_b.result() == 8
+
+
+@parsl.python_app
+def make_path(s: str):
+    return Path(s)
+
+
+@parsl.python_app
+def append_paths(iterable: Iterable[Path], end_str: str = "end"):
+    type_ = type(iterable)
+    return type_([Path(s, end_str) for s in iterable])
```

Comment:
> parsl/tests/test_python_apps/test_pluggable_future_resolution.py:54: error: Too many arguments for "Iterable"  [call-arg]

there's no static guarantee that the constructor of an arbitrary type that happens to be an iterable takes one parameter that is a list (of things you want to somehow populate that iterable with)

more technically: this is coming from (I think) because we statically know that `iterable` is an Iterable, so _type = Iterable at least as far as finding a single value that always works, and then type_(....) is Iterable(....) but Iterable() takes no parameters.

Reply:
Makes sense. This was me trying to be too smart, I suppose...